app/js/lib/foxbox/foxbox.js

```diff
@@ -425,97 +434,54 @@ export default class Foxbox extends Service {
   }
 
   /**
-   * Ask the user for acepting push notifications from the box.
+   * Ask the user for accepting push notifications from the box.
    * This method will be call each time that we log in, but will
-   * stop the execution if we already have the push subcription
+   * stop the execution if we already have the push subscription
    * information.
    *
    * @param {boolean} resubscribe Parameter used for testing
-   * purposes, and follow the whole subscrition process even if
+   * purposes, and follow the whole subscription process even if
    * we have push subscription information.
    */
   subscribeToNotifications(resubscribe = false) {
     return this[p.webPush].subscribeToNotifications(resubscribe);
   }
 
-  performSetOperation(operation, value) {
-    let operationType = this[p.getOperationValueType](operation.kind);
+  [p.fetchServices]() {
     return this[p.net].fetchJSON(
-      `${this[p.net].origin}/api/v${this[p.settings].apiVersion}/channels/set`,
-      'PUT',
-      // Query operation by id.
-      [[{ id: operation.id }, { [operationType]: value }]]
-    );
+      `${this[p.net].origin}/api/v${this[p.settings].apiVersion}/services`
+      )
+      .then((services) => services.map((service) => {
+        // Some service don't have name, but can have product_name instead,
+        // eventually we should get rid of this line and use unified service
+        // label field provided by taxonomy.
+        service.properties.name = service.properties.name ||
+          service.properties.product_name;
+
+        return service;
+      }));
   }
 
-  performGetOperation(operation) {
-    let payload = { id: operation.id };
-
-    if (operation.kind.type === 'Binary') {
-      return this[p.net].fetchBlob(
-        `${this[p.net].origin}/api/v${this[p.settings].apiVersion}` +
-        '/channels/get',
-        // For now we only support JPEG blobs.
-        'image/jpeg',
-        'PUT',
-        payload
-      );
-    }
+  [p.getServiceInstance](data) {
+    const config = {
+      net: this[p.net],
+      settings: this[p.settings]
+    };
 
-    return this[p.net].fetchJSON(
-      `${this[p.net].origin}/api/v${this[p.settings].apiVersion}/channels/get`,
-      'PUT',
-      payload
-    );
-  }
+    switch (data.adapter) {
+      case 'thinkerbell-adapter':
+      case 'webpush@link.mozilla.org':
+      case 'console@link.mozilla.org':
+        return data;
```

Comment:
Yes, you're right that's totally inconsistent, but these blacklisted adapters shouldn't be exposed to the user anyway. I remember we had this discussion about whether the filtering of unused adapters should take place in the view or in the lib. You did it in the view but I still think it should happen in the lib.
I suggest to continue doing it in the view for now, but in the future I'd really like to reconsider this option, as long as these adapters won't be used in the view or exposed to the user.
What do you think?